homeassistant/components/sensor/rfxtrx.py

```diff
@@ -29,25 +31,44 @@ def setup_platform(hass, config, add_devices_callback, discovery_info=None):
     """Setup the RFXtrx platform."""
     from RFXtrx import SensorEvent
 
+    sensors = []
+    for entity_id, entity_info in config.get('devices', {}).items():
+        if entity_id not in rfxtrx.RFX_DEVICES:
+            _LOGGER.info("Add %s rfxtrx.sensor", entity_info[ATTR_NAME])
+            event = rfxtrx.get_rfx_object(entity_info[ATTR_PACKETID])
+            new_sensor = RfxtrxSensor(event, entity_info[ATTR_NAME])
+            rfxtrx.RFX_DEVICES[entity_id] = new_sensor
+            sensors.append(new_sensor)
+
+    add_devices_callback(sensors)
+
     def sensor_update(event):
         """Callback for sensor updates from the RFXtrx gateway."""
-        if isinstance(event, SensorEvent):
-            entity_id = slugify(event.device.id_string.lower())
-
-            # Add entity if not exist and the automatic_add is True
-            if entity_id not in rfxtrx.RFX_DEVICES:
-                automatic_add = config.get('automatic_add', True)
-                if automatic_add:
-                    _LOGGER.info("Automatic add %s rfxtrx.sensor", entity_id)
-                    new_sensor = RfxtrxSensor(event)
-                    rfxtrx.RFX_DEVICES[entity_id] = new_sensor
-                    add_devices_callback([new_sensor])
-            else:
-                _LOGGER.debug(
-                    "EntityID: %s sensor_update",
+        if not isinstance(event, SensorEvent):
+            return
+
+        entity_id = "s" + slugify(event.device.id_string.lower())
+
+        # Add entity if not exist and the automatic_add is True
+        if entity_id not in rfxtrx.RFX_DEVICES:
+            automatic_add = config.get('automatic_add', True)
+            if automatic_add:
+                pkt_id = "".join("{0:02x}".format(x) for x in event.data)
+                entity_name = "%s : %s" % (entity_id, pkt_id)
+                _LOGGER.info(
+                    "Automatic add rfxtrx.sensor: (%s : %s)",
                     entity_id,
-                )
-                rfxtrx.RFX_DEVICES[entity_id].event = event
+                    pkt_id)
+
+                new_sensor = RfxtrxSensor(event, entity_name)
+                rfxtrx.RFX_DEVICES[entity_id] = new_sensor
+                add_devices_callback([new_sensor])
+        else:
```

Comment:
You can rewrite this as a guard clause, put it above the current if-statement and remove the if-statement:
```python
if device_id in rfxtrx.RFX_DEVICES:
    rfxtrx.RFX_DEVICES[device_id].event = event
    return
```